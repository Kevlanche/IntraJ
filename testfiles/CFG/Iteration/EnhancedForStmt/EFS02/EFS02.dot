digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[6]"}
{ rank = same; "TypeImportOnDemandDecl[4]"; "ClassDecl[7]"}
{ rank = same; "PackageAccess[5]"; "Modifiers[8]"; "Opt[10]"; "List[11]"; "List[12]"}
{ rank = same; "List[9]"; "MethodDecl[13]"}
{ rank = same; "Entry[14]"; "Modifiers[15]"; "PrimitiveTypeAccess[17]"; "List[18]"; "List[19]"; "Opt[20]"; "Exit[21]"}
{ rank = same; "List[16]"; "Block[21]"}
{ rank = same; "List[22]"}
{ rank = same; "VarDeclStmt[23]"; "EnhancedForStmt[43]"; "VarDeclStmt[84]"}
{ rank = same; "Modifiers[24]"; "ParTypeAccess[26]"; "List[30]"; "Modifiers[44]"; "TypeAccess[46]"; "VariableDeclarator[47]"; "VarAccess[50]"; "Block[51]"; "Modifiers[85]"; "PrimitiveTypeAccess[87]"; "List[88]"}
{ rank = same; "List[25]"; "TypeAccess[27]"; "List[28]"; "VariableDeclarator[31]"; "List[45]"; "List[48]"; "Opt[49]"; "List[52]"; "List[86]"; "VariableDeclarator[89]"}
{ rank = same; "TypeAccess[29]"; "List[32]"; "Opt[33]"; "VarDeclStmt[53]"; "IfStmt[62]"; "VarDeclStmt[75]"; "List[90]"; "Opt[91]"}
{ rank = same; "ClassInstanceExpr[34]"; "Modifiers[54]"; "PrimitiveTypeAccess[56]"; "List[57]"; "Dot[63]"; "ContinueStmt[71]"; "Opt[74]"; "Modifiers[76]"; "PrimitiveTypeAccess[78]"; "List[79]"}
{ rank = same; "ParTypeAccess[35]"; "List[39]"; "Opt[40]"; "List[41]"; "List[42]"; "List[55]"; "VariableDeclarator[58]"; "VarAccess[64]"; "MethodAccess[65]"; "List[73]"; "UncheckedExceptions[72]"; "List[77]"; "VariableDeclarator[80]"}
{ rank = same; "TypeAccess[36]"; "List[37]"; "List[59]"; "Opt[60]"; "List[67]"; "List[69]"; "List[70]"; "UncheckedExceptions[66]"; "List[81]"; "Opt[82]"}
{ rank = same; "TypeAccess[38]"; "IntegerLiteral[61]"; "StringLiteral[68]"; "IntegerLiteral[83]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeImportOnDemandDecl[4]" [label="TypeImportOnDemandDecl\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PackageAccess[5]" [label="PackageAccess\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[6]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[7]" [label="ClassDecl\n < EFS02 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[8]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[9]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[10]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[12]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[13]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[14]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[15]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[16]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[17]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[18]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[19]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[20]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[21]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[22]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[23]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[24]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[25]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParTypeAccess[26]" [label="ParTypeAccess\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[27]" [label="TypeAccess\n < Collection >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[28]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[29]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[31]" [label="VariableDeclarator\n < Collection<String> coll >", fillcolor=white   style=filled  ]
"List[32]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[33]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassInstanceExpr[34]" [label="ClassInstanceExpr\n <LinkedList>", fillcolor=white   style=filled  ]
"ParTypeAccess[35]" [label="ParTypeAccess\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[36]" [label="TypeAccess\n < LinkedList >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[38]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[39]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[40]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[41]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"EnhancedForStmt[43]" [label="EnhancedForStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[44]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[45]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[46]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[47]" [label="VariableDeclarator\n < String str >", fillcolor=white   style=filled  ]
"List[48]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[49]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[50]" [label="VarAccess\n < coll >", fillcolor=white   style=filled  ]
"Block[51]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[52]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[53]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[54]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[55]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[56]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[57]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[58]" [label="VariableDeclarator\n < int a >", fillcolor=white   style=filled  ]
"List[59]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[60]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[61]" [label="IntegerLiteral\n < 0 >", fillcolor=white   style=filled  ]
"IfStmt[62]" [label="IfStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[63]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[64]" [label="VarAccess\n < str >", fillcolor=white   style=filled  ]
"MethodAccess[65]" [label="MethodAccess\n < equals >", fillcolor=white   style=filled  ]
"List[67]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[68]" [label="StringLiteral\n <  >", fillcolor=white   style=filled  ]
"List[69]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[70]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[66]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ContinueStmt[71]" [label="ContinueStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[73]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[72]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[74]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[75]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[76]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[77]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[78]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[79]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[80]" [label="VariableDeclarator\n < int b >", fillcolor=white   style=filled  ]
"List[81]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[82]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[83]" [label="IntegerLiteral\n < 0 >", fillcolor=white   style=filled  ]
"VarDeclStmt[84]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[85]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[86]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[87]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[88]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[89]" [label="VariableDeclarator\n < int c >", fillcolor=white   style=filled  ]
"List[90]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[91]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[21]" [label="Exit\n ", fillcolor=white   style=filled  ]
"TypeImportOnDemandDecl[4]" -> "PackageAccess[5]" [style=dashed, color=gray]
"List[3]" -> "TypeImportOnDemandDecl[4]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"Modifiers[8]" -> "List[9]" [style=dashed, color=gray]
"ClassDecl[7]" -> "Modifiers[8]" [style=dashed, color=gray]
"ClassDecl[7]" -> "Opt[10]" [style=dashed, color=gray]
"ClassDecl[7]" -> "List[11]" [style=dashed, color=gray]
"MethodDecl[13]" -> "Entry[14]" [style=dashed, color=gray]
"Modifiers[15]" -> "List[16]" [style=dashed, color=gray]
"MethodDecl[13]" -> "Modifiers[15]" [style=dashed, color=gray]
"MethodDecl[13]" -> "PrimitiveTypeAccess[17]" [style=dashed, color=gray]
"MethodDecl[13]" -> "List[18]" [style=dashed, color=gray]
"MethodDecl[13]" -> "List[19]" [style=dashed, color=gray]
"Modifiers[24]" -> "List[25]" [style=dashed, color=gray]
"VarDeclStmt[23]" -> "Modifiers[24]" [style=dashed, color=gray]
"ParTypeAccess[26]" -> "TypeAccess[27]" [style=dashed, color=gray]
"List[28]" -> "TypeAccess[29]" [style=dashed, color=gray]
"ParTypeAccess[26]" -> "List[28]" [style=dashed, color=gray]
"VarDeclStmt[23]" -> "ParTypeAccess[26]" [style=dashed, color=gray]
"VariableDeclarator[31]" -> "List[32]" [style=dashed, color=gray]
"ParTypeAccess[35]" -> "TypeAccess[36]" [style=dashed, color=gray]
"List[37]" -> "TypeAccess[38]" [style=dashed, color=gray]
"ParTypeAccess[35]" -> "List[37]" [style=dashed, color=gray]
"ClassInstanceExpr[34]" -> "ParTypeAccess[35]" [style=dashed, color=gray]
"ClassInstanceExpr[34]" -> "List[39]" [style=dashed, color=gray]
"ClassInstanceExpr[34]" -> "Opt[40]" [style=dashed, color=gray]
"ClassInstanceExpr[34]" -> "List[41]" [style=dashed, color=gray]
"ClassInstanceExpr[34]" -> "List[42]" [style=dashed, color=gray]
"Opt[33]" -> "ClassInstanceExpr[34]" [style=dashed, color=gray]
"VariableDeclarator[31]" -> "Opt[33]" [style=dashed, color=gray]
"List[30]" -> "VariableDeclarator[31]" [style=dashed, color=gray]
"VarDeclStmt[23]" -> "List[30]" [style=dashed, color=gray]
"List[22]" -> "VarDeclStmt[23]" [style=dashed, color=gray]
"Modifiers[44]" -> "List[45]" [style=dashed, color=gray]
"EnhancedForStmt[43]" -> "Modifiers[44]" [style=dashed, color=gray]
"EnhancedForStmt[43]" -> "TypeAccess[46]" [style=dashed, color=gray]
"VariableDeclarator[47]" -> "List[48]" [style=dashed, color=gray]
"VariableDeclarator[47]" -> "Opt[49]" [style=dashed, color=gray]
"EnhancedForStmt[43]" -> "VariableDeclarator[47]" [style=dashed, color=gray]
"EnhancedForStmt[43]" -> "VarAccess[50]" [style=dashed, color=gray]
"Modifiers[54]" -> "List[55]" [style=dashed, color=gray]
"VarDeclStmt[53]" -> "Modifiers[54]" [style=dashed, color=gray]
"VarDeclStmt[53]" -> "PrimitiveTypeAccess[56]" [style=dashed, color=gray]
"VariableDeclarator[58]" -> "List[59]" [style=dashed, color=gray]
"Opt[60]" -> "IntegerLiteral[61]" [style=dashed, color=gray]
"VariableDeclarator[58]" -> "Opt[60]" [style=dashed, color=gray]
"List[57]" -> "VariableDeclarator[58]" [style=dashed, color=gray]
"VarDeclStmt[53]" -> "List[57]" [style=dashed, color=gray]
"List[52]" -> "VarDeclStmt[53]" [style=dashed, color=gray]
"Dot[63]" -> "VarAccess[64]" [style=dashed, color=gray]
"List[67]" -> "StringLiteral[68]" [style=dashed, color=gray]
"MethodAccess[65]" -> "List[67]" [style=dashed, color=gray]
"MethodAccess[65]" -> "List[69]" [style=dashed, color=gray]
"MethodAccess[65]" -> "List[70]" [style=dashed, color=gray]
"MethodAccess[65]" -> "UncheckedExceptions[66]" [style=dashed, color=gray]
"Dot[63]" -> "MethodAccess[65]" [style=dashed, color=gray]
"IfStmt[62]" -> "Dot[63]" [style=dashed, color=gray]
"ContinueStmt[71]" -> "List[73]" [style=dashed, color=gray]
"ContinueStmt[71]" -> "UncheckedExceptions[72]" [style=dashed, color=gray]
"IfStmt[62]" -> "ContinueStmt[71]" [style=dashed, color=gray]
"IfStmt[62]" -> "Opt[74]" [style=dashed, color=gray]
"List[52]" -> "IfStmt[62]" [style=dashed, color=gray]
"Modifiers[76]" -> "List[77]" [style=dashed, color=gray]
"VarDeclStmt[75]" -> "Modifiers[76]" [style=dashed, color=gray]
"VarDeclStmt[75]" -> "PrimitiveTypeAccess[78]" [style=dashed, color=gray]
"VariableDeclarator[80]" -> "List[81]" [style=dashed, color=gray]
"Opt[82]" -> "IntegerLiteral[83]" [style=dashed, color=gray]
"VariableDeclarator[80]" -> "Opt[82]" [style=dashed, color=gray]
"List[79]" -> "VariableDeclarator[80]" [style=dashed, color=gray]
"VarDeclStmt[75]" -> "List[79]" [style=dashed, color=gray]
"List[52]" -> "VarDeclStmt[75]" [style=dashed, color=gray]
"Block[51]" -> "List[52]" [style=dashed, color=gray]
"EnhancedForStmt[43]" -> "Block[51]" [style=dashed, color=gray]
"List[22]" -> "EnhancedForStmt[43]" [style=dashed, color=gray]
"Modifiers[85]" -> "List[86]" [style=dashed, color=gray]
"VarDeclStmt[84]" -> "Modifiers[85]" [style=dashed, color=gray]
"VarDeclStmt[84]" -> "PrimitiveTypeAccess[87]" [style=dashed, color=gray]
"VariableDeclarator[89]" -> "List[90]" [style=dashed, color=gray]
"VariableDeclarator[89]" -> "Opt[91]" [style=dashed, color=gray]
"List[88]" -> "VariableDeclarator[89]" [style=dashed, color=gray]
"VarDeclStmt[84]" -> "List[88]" [style=dashed, color=gray]
"List[22]" -> "VarDeclStmt[84]" [style=dashed, color=gray]
"Block[21]" -> "List[22]" [style=dashed, color=gray]
"Opt[20]" -> "Block[21]" [style=dashed, color=gray]
"MethodDecl[13]" -> "Opt[20]" [style=dashed, color=gray]
"MethodDecl[13]" -> "Exit[21]" [style=dashed, color=gray]
"List[12]" -> "MethodDecl[13]" [style=dashed, color=gray]
"ClassDecl[7]" -> "List[12]" [style=dashed, color=gray]
"List[6]" -> "ClassDecl[7]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[6]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"VariableDeclarator[80]" -> "VarAccess[50]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[80]" -> "IntegerLiteral[83]" [color=red, constraint=false]
"IntegerLiteral[83]" -> "VariableDeclarator[80]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[83]" -> "MethodAccess[65]" [color=red, constraint=false]
"MethodAccess[65]" -> "VarAccess[50]" [color=blue, constraint=false, xlabel="T "] 
"MethodAccess[65]" -> "IntegerLiteral[83]" [color=blue, constraint=false, xlabel="F "] 
"MethodAccess[65]" -> "StringLiteral[68]" [color=red, constraint=false]
"StringLiteral[68]" -> "MethodAccess[65]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[68]" -> "VarAccess[64]" [color=red, constraint=false]
"VarAccess[64]" -> "StringLiteral[68]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[64]" -> "VariableDeclarator[58]" [color=red, constraint=false]
"VariableDeclarator[58]" -> "VarAccess[64]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[58]" -> "IntegerLiteral[61]" [color=red, constraint=false]
"IntegerLiteral[61]" -> "VariableDeclarator[58]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[61]" -> "VariableDeclarator[47]" [color=red, constraint=false]
"VariableDeclarator[47]" -> "IntegerLiteral[61]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[47]" -> "VarAccess[50]" [color=red, constraint=false]
"Exit[21]" -> "VariableDeclarator[89]" [color=red, constraint=false]
"VariableDeclarator[89]" -> "Exit[21]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[89]" -> "VarAccess[50]" [color=red, constraint=false]
"VarAccess[50]" -> "VariableDeclarator[47]" [color=blue, constraint=false, xlabel="T "] 
"VarAccess[50]" -> "VariableDeclarator[89]" [color=blue, constraint=false, xlabel="F "] 
"VarAccess[50]" -> "VariableDeclarator[31]" [color=red, constraint=false]
"VarAccess[50]" -> "MethodAccess[65]" [color=red, constraint=false]
"VarAccess[50]" -> "VariableDeclarator[80]" [color=red, constraint=false]
"VariableDeclarator[31]" -> "VarAccess[50]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[31]" -> "ClassInstanceExpr[34]" [color=red, constraint=false]
"ClassInstanceExpr[34]" -> "VariableDeclarator[31]" [color=blue, constraint=false, xlabel=" "] 
"ClassInstanceExpr[34]" -> "Entry[14]" [color=red, constraint=false]
"Entry[14]" -> "ClassInstanceExpr[34]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[6]" [style=invis]
"Modifiers[8]" -> "Opt[10]" -> "List[11]" -> "List[12]" [style=invis]
"Entry[14]" -> "Modifiers[15]" -> "PrimitiveTypeAccess[17]" -> "List[18]" -> "List[19]" -> "Opt[20]" -> "Exit[21]" [style=invis]
"VarDeclStmt[23]" -> "EnhancedForStmt[43]" -> "VarDeclStmt[84]" [style=invis]
"Modifiers[24]" -> "ParTypeAccess[26]" -> "List[30]" [style=invis]
"TypeAccess[27]" -> "List[28]" [style=invis]
"List[32]" -> "Opt[33]" [style=invis]
"ParTypeAccess[35]" -> "List[39]" -> "Opt[40]" [style=invis]
"TypeAccess[36]" -> "List[37]" [style=invis]
"Modifiers[44]" -> "TypeAccess[46]" -> "VariableDeclarator[47]" -> "VarAccess[50]" -> "Block[51]" [style=invis]
"List[48]" -> "Opt[49]" [style=invis]
"VarDeclStmt[53]" -> "IfStmt[62]" -> "VarDeclStmt[75]" [style=invis]
"Modifiers[54]" -> "PrimitiveTypeAccess[56]" -> "List[57]" [style=invis]
"List[59]" -> "Opt[60]" [style=invis]
"Dot[63]" -> "ContinueStmt[71]" -> "Opt[74]" [style=invis]
"VarAccess[64]" -> "MethodAccess[65]" [style=invis]
"Modifiers[76]" -> "PrimitiveTypeAccess[78]" -> "List[79]" [style=invis]
"List[81]" -> "Opt[82]" [style=invis]
"Modifiers[85]" -> "PrimitiveTypeAccess[87]" -> "List[88]" [style=invis]
"List[90]" -> "Opt[91]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
93", fillcolor=1]
Edges [label="Total Nr. Edges
122", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
14", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
15", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}