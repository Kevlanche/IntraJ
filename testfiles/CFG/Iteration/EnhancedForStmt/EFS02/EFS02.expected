NodeName: Entry[14] Successors: [ClassInstanceExpr[34]]
NodeName: ClassInstanceExpr[34] Successors: [VariableDeclarator[31]]
NodeName: VariableDeclarator[31] Successors: [VarAccess[50]]
NodeName: VarAccess[50] Successors: [VariableDeclarator[47], VariableDeclarator[89]]
NodeName: VariableDeclarator[47] Successors: [IntegerLiteral[61]]
NodeName: IntegerLiteral[61] Successors: [VariableDeclarator[58]]
NodeName: VariableDeclarator[58] Successors: [VarAccess[64]]
NodeName: VarAccess[64] Successors: [StringLiteral[68]]
NodeName: StringLiteral[68] Successors: [MethodAccess[65]]
NodeName: MethodAccess[65] Successors: [VarAccess[50], IntegerLiteral[83]]
NodeName: IntegerLiteral[83] Successors: [VariableDeclarator[80]]
NodeName: VariableDeclarator[80] Successors: [VarAccess[50]]
NodeName: VariableDeclarator[89] Successors: [Exit[21]]
NodeName: Exit[21] Successors: []

NodeName: Exit[21] Predecessor: [VariableDeclarator[89]]
NodeName: VariableDeclarator[89] Predecessor: [VarAccess[50]]
NodeName: VarAccess[50] Predecessor: [VariableDeclarator[31], MethodAccess[65], VariableDeclarator[80]]
NodeName: VariableDeclarator[31] Predecessor: [ClassInstanceExpr[34]]
NodeName: ClassInstanceExpr[34] Predecessor: [Entry[14]]
NodeName: Entry[14] Predecessor: []
NodeName: MethodAccess[65] Predecessor: [StringLiteral[68]]
NodeName: StringLiteral[68] Predecessor: [VarAccess[64]]
NodeName: VarAccess[64] Predecessor: [VariableDeclarator[58]]
NodeName: VariableDeclarator[58] Predecessor: [IntegerLiteral[61]]
NodeName: IntegerLiteral[61] Predecessor: [VariableDeclarator[47]]
NodeName: VariableDeclarator[47] Predecessor: [VarAccess[50]]
NodeName: VariableDeclarator[80] Predecessor: [IntegerLiteral[83]]
NodeName: IntegerLiteral[83] Predecessor: [MethodAccess[65]]
